### PR TITLE
'done'

### PR DESCRIPTION
**Pull request by Hanna-Balabukha**

### CODE REVIEW

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,48 @@
...
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  dead() {
+    const indexAnimal = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1)
+    }
```
Comment (loralevitska) [inline src/herbivoresAndCarnivores.js:16]: Remove redundant unused method

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,48 @@
...
+  constructor(name, health = 100, hidden = false) {
+    super(name, health, hidden);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+    }
```
Comment (loralevitska) [inline src/herbivoresAndCarnivores.js:30]:
```suggestion
   this.hidden = true;
```

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -1,15 +1,48 @@
...
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(animal) {
+    if (animal.hidden === false) {
```
Comment (loralevitska) [inline src/herbivoresAndCarnivores.js:40]: Hidden is boolean variable so check it like booleans
Check animal is it Herbivore or Carnivore
```suggestion
    if (animal instanceof Herbivore && !herbivore.hidden) {
```

File: src/herbivoresAndCarnivores.js [hunk 4]
```diff
@@ -1,15 +1,48 @@
...
+      animal.health -= 50;
+    }
+
+   Animal.alive = Animal.alive.filter(x => x.health !== 0)
```
Comment (loralevitska) [inline src/herbivoresAndCarnivores.js:44]: What is `x`? Try to use more meaningful variable names `beast` `creature` etc.

**Review by tetiana-ivanchenko — APPROVED** [review]

**Pull request closed without merging**